enhancements/local-storage/automatic-detection.md

```diff
@@ -64,40 +65,299 @@ and we can use that information to sort devices into storage classes based on th
+  DiscoveryPending DiscoveryPhase = "discovering"
+  // DiscoveryFailed represents that the discovery process has failed
+  DiscoveryFailed DiscoveryPhase = "error"
+)
+
+type LocalVolumeDiscoveryStatus struct {
+       // Phase represents the current phase of discovery process
+       // This is used by the OLM UI to provide status information
+       // to the user
+	Phase DiscoveryPhase `json:"phase,omitempty"`
+	// Conditions is a list of conditions and their status.
+	Conditions []operatorv1.OperatorCondition `json:"conditions,omitempty"`
+	// observedGeneration is the last generation change the operator has dealt with
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+}
+
+type LocalVolumeDiscoverySpec struct {
+	// Nodes on which the automatic detection policies must run.
+	// +optional
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// If specified tolerations is the list of toleration that is passed to the
+	// LocalVolumeDiscovery Daemon
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type LocalVolumeDiscovery struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              LocalVolumeDiscoverySpec   `json:"spec"`
+	Status            LocalVolumeDiscoveryStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeDiscoveryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeDiscovery `json:"items"`
+}
+```
+
+API scheme for `LocalVolumeDiscoveryResult` CR:
+```go
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational or nonRotational
+type DeviceMechanicalProperty string
+
+// The mechanical properties of the devices
+const (
+	// Rotational refers to magnetic disks
+	Rotational DeviceMechanicalProperty = "Rotational"
+	// NonRotational refers to ssds
+	NonRotational DeviceMechanicalProperty = "NonRotational"
+)
+
+// DeviceType represents the type of devices that would be discoverd.
+type DeviceType string
+
+// Devices types that should be discovered
+const (
+	// RawDisk represents a device-type of block disk
+	RawDisk DeviceType = "disk"
+	// Loop type device
+	Loop DeviceType = "loop"
+)
+
+type DiscoveredDevice struct {
+	// Name represents the persitent name of the device. For eg, /dev/disk/by-id/...
+	Name string `json:"name"`
+	// Model of the detected device
+	Model string `json:"model,omitempty"`
+	// Type of the detected device
+  Type DeviceType `json:"type"`
+	// Vendor of the detected device
+	Vendor string `json:"vendor,omitempty"`
+	// Size of the detected device
+	Size resource.Quantity `json:"size,omitempty"`
+	// Property represents whether the device type is rotational or not
+	Property DeviceMechanicalProperty `json:"property"`
+        // FSType represents the filesystem available on the device
+	FSType string `json:"fstype",omitempty`
+}
```

Comment:
How about some flag to indicate if the device is still available for a PV? After a local PV is created on the device, it should no longer be considered available.

Reply:
Added the `Status` field in the `DiscoveryDevices`. This will be object having information like `state` (whether the device is available or in-use), `provisionedPV` (any existing provisioned persistent volume on the disk) and `storageclass` (storageclass of the provisioned persistent volume). 

Let me know what you think about these changes @travisn .  I'm looking into more scenarios when the device can be marked as `in-use`